distribution/docs/src/main/resources/_contents/_catalog-frameworks/developing-catalog-frameworks-contents.adoc

```diff
@@ -0,0 +1,192 @@
+
+.Example Query Methods
+[source,java,linenums]
+----
+public QueryResponse query(QueryRequest query) throws UnsupportedQueryException,SourceUnavailableException, FederationException;
+public QueryResponse query(QueryRequest queryRequest, FederationStrategy strategy) throws SourceUnavailableException, UnsupportedQueryException, FederationException;
+----
+Query requests process  `PolicyPlugin`, `AccessPlugin`, and `PreQueryPlugin` instances before execution and  `PolicyPlugin`, `AccessPlugin`, and `PostQueryPlugin` instances after execution.
+
+====== Resource Methods
+
+Resource methods retrieve products from Sources.
+
+.Example Resource Methods
+[source,java,linenums]
+----
+public ResourceResponse getEnterpriseResource(ResourceRequest request) throwsIOException, ResourceNotFoundException, ResourceNotSupportedException;
+public ResourceResponse getLocalResource(ResourceRequest request) throws IOException, ResourceNotFoundException, ResourceNotSupportedException;
+public ResourceResponse getResource(ResourceRequest request, String resourceSiteName) throws IOException, ResourceNotFoundException, ResourceNotSupportedException;
+----
+Resource requests process `PreResourcePlugin`s before execution and `PostResourcePlugin`s after execution.
+
+====== Source Methods
+
+Source methods can get a list of Source identifiers or request descriptions about Sources.
+
+.Example Source Methods
+[source,java,linenums]
+----
+public Set<String> getSourceIds();
+public SourceInfoResponse getSourceInfo(SourceInfoRequest sourceInfoRequest) throws SourceUnavailableException;
+----
+
+====== Transform Methods
+
+Transform methods provide convenience methods for using Metacard Transformers and Query Response Transformers.
+
+.Transform Methods
+[source,java,linenums]
+----
+// Metacard Transformer
+public BinaryContent transform(Metacard metacard, String transformerId, Map<String,Serializable> requestProperties) throws CatalogTransformerException;
+
+// Query Response Transformer
+public BinaryContent transform(SourceResponse response, String transformerId, Map<String, Serializable> requestProperties) throws CatalogTransformerException;
+----
+
+.[[_included_catalog_frameworks]]Included Catalog Frameworks
+<<_catalog_api,Catalog API>>:: description
+
+===== Implementing Catalog Methods
+
+.Query Response Transform Example
+[source,java,linenums]
+----
+// inject CatalogFramework instance or retrieve an instance
+private CatalogFramework catalogFramework;
+
+public RSSEndpoint(CatalogFramework catalogFramework)
+{
+     this.catalogFramework = catalogFramework ;
+     // implementation
+}
+
+ // Other implementation details ...
+
+private void convert(QueryResponse queryResponse ) {
+    // ...
+    String transformerId = "rss";
+
+    BinaryContent content = catalogFramework.transform(queryResponse, transformerId, null);
+
+    // ...
+
+}
+----
+
+[cols="2,5" options="header"]
+|===
+|Line #
```

Comment:
Agreed. I'm not a fan of referencing the line numbers in these examples for that and other reasons.